src/workerd/api/base64.c++

```diff
@@ -6,35 +6,38 @@
 
 namespace workerd::api {
 
-kj::Array<kj::byte> Base64Module::decodeArray(kj::Array<kj::byte> input) {
-  auto size = simdutf::maximal_binary_length_from_base64((const char*)input.begin(), input.size());
-  auto buf = kj::heapArray<kj::byte>(size);
-  auto result = simdutf::base64_to_binary(
-      input.asChars().begin(), input.size(), buf.asChars().begin(), simdutf::base64_default);
+jsg::BufferSource Base64Module::decodeArray(jsg::Lock& js, jsg::BufferSource input) {
+  auto ptr = input.asArrayPtr();
+  auto size = simdutf::maximal_binary_length_from_base64((const char*)ptr.begin(), ptr.size());
+  auto buf = jsg::BackingStore::alloc<v8::ArrayBuffer>(js, size);
+  auto result = simdutf::base64_to_binary(ptr.asChars().begin(), input.size(),
+      buf.asArrayPtr().asChars().begin(), simdutf::base64_default);
   JSG_REQUIRE(result.error == simdutf::SUCCESS, DOMSyntaxError,
       kj::str("Invalid base64 at position ", result.count, ": ",
           simdutf::error_to_string(result.error).data()));
   KJ_ASSERT(result.count <= size);
-  return buf.slice(0, result.count).attach(kj::mv(buf));
+  buf.trim(buf.size() - result.count);
+  return jsg::BufferSource(js, kj::mv(buf));
 }
 
-kj::Array<kj::byte> Base64Module::encodeArray(kj::Array<kj::byte> input) {
+jsg::BufferSource Base64Module::encodeArray(jsg::Lock& js, jsg::BufferSource input) {
   auto size = simdutf::base64_length_from_binary(input.size());
-  auto buf = kj::heapArray<kj::byte>(size);
-  auto out_size = simdutf::binary_to_base64(
-      input.asChars().begin(), input.size(), buf.asChars().begin(), simdutf::base64_default);
+  auto buf = jsg::BackingStore::alloc<v8::ArrayBuffer>(js, size);
+  auto out_size = simdutf::binary_to_base64(input.asArrayPtr().asChars().begin(), input.size(),
+      buf.asArrayPtr().asChars().begin(), simdutf::base64_default);
   KJ_ASSERT(out_size <= size);
-  return buf.slice(0, out_size).attach(kj::mv(buf));
+  buf.trim(size - out_size);
```

Comment:
This is kind of clunky. I see a `limit` method in BackingStore - could you use that instead?